sample_formatted_code.do_not_merge.c

```diff
@@ -0,0 +1,562 @@
+{
+    // multiple consecutive vars (comments and vars can be aligned)
+    // AlignConsecutiveAssignments
+    // AlignConsecutiveDeclarations
+    // AlignTrailingComments
+    int a = 13;           /* comment */
+    int32_t abc = 1312;   /* comment */
+    int abcdefghikl = 13; /* comment */
+
+    // AlwaysBreakBeforeMultilineStrings
+    aaaa = "bbbb"
+           "ccc";
+
+    //--- variable init continuation behaviour
+    // Cpp11BracedListStyle impacts if space at beginning and end of brace
+    // e.g. false: Bla bla[] = { 1, 2, 3 };
+    //      true:  Bla bla[] = {1, 2, 3};
+    // Note, the different indentation of continuation line:
+    //  - Cpp11BracedListStyle:false does NOT use ContinuationIndentWidth (8)
+    //    but rather the regular IndentWidth (4)! Bug? Feature?
+    //  - Cpp11BracedListStyle:true uses ContinuationIndentWidth (8)
+    int list[] = {1, 2, 3, 4, 5, 6, 7, 8, 9};
+    int list[] = {
+            1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18};
+    int list[] = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18,
+            19, 20, 21, 22};
+    PcapStats64 last = {0, 1234, 8765};
+    // trailing comma forces one-value-by-line
+    PcapStats64 trailing_comma = {
+            0,
+            1234,
+            8765,
+    };
+    struct pcap_stat current = {12, 2345, 9876, 1345};
+    struct pcap_stat current = {
+            12, 2345, 9876, 1345, 333, 444, 5555, 66666, 7777777};
+    current = {12, 2345, 9876, 1345, 333, 444, 5555, 66666, 7777777, 888888,
+            9999999};
+
+    //--- designated initializer continuation behaviour
+    // Current clang-format disables BinPacking for designated intializers when
+    // continuing on more than one line.
+    PcapStats64 last = {.ps_recv = 0, .ps_drop = 1234, .ps_ifdrop = 8765};
+    // Cpp11BracedListStyle:false puts end brace onto separate line iff
+    // continuation line can hold all intializers. Bug? Feature?
+    struct pcap_stat current = {
+            .ps_recv = 12, .ps_drop = 2345, .ps_ifdrop = 9876, .ps_what = 134};
+    pcap_stat current = {
+            .ps_recv = 12, .ps_drop = 2345, .ps_ifdrop = 9876, .ps_what = 134};
+    // One designated initializer per line if it does not fit into one
+    // continuation line as BinPacking is disabled for designated intializer.
+    struct pcap_stat current = {.ps_recv = 12,
+            .ps_drop = 2345,
+            .ps_ifdrop = 9876,
+            .ps_what = 1345,
+            .ps_more = 333};
+    current = {.ps_recv = 12,
+            .ps_drop = 2345,
+            .ps_ifdrop = 9876,
+            .ps_what = 1345,
+            .ps_more = 333};
+
+    // function call continuation
+    function_call(
+            with, many, params, that, will spill, over, eventually, iff, max);
+    function_call(with, many, params, that, will spill, over, eventually, iff,
+            we, keep, on, adding);
+}
+
+struct Bitfields {
+    // AlignConsecutiveBitFields
+    int aaaa : 1;
```

Comment:
There will be spaces around the colon for bitfields which is different than used in e.g. app-layer-dnp3-objects.h

Reply:
Is the comment ```// AlignConsecutiveBitFields``` required here? What would happen if left off?

Reply:
No, sorry. It's my poor way of tracking that "this is the clang-format setting that impacts this". None of the // comments are required, just FYI. These settings are only set in the .clang-format file.

I am currently updating the code_style.rst where it should be easier to follow.

I was trying to avoid including a "this vs that" comparison for each setting, but it probably would be helpful for the bikeshedding discussion on the format. ;)

Reply:
AlignConsecutiveBitFields (bool)
If true, aligns consecutive bitfield members.

This will align the bitfield separators of consecutive lines. This will result in formattings like

```
int aaaa : 1;
int b    : 12;
int ccc  : 8;
```

Oh, this is only supported as of clang 11 which is not yet out. I don't think this should be set at all then. Explains why you won't find this setting in clang-format.full.do_not_merge. I've used clang 9 which is recent enough yet old enough to be supported by different dists.